### DISCUSSION TITLE
Are sessions really needed?

### DISCUSSION THREAD

**Discussion by CyberTailor** [benbusby/whoogle-search#788] [Ideas]
Whoogle can store configuration in plain cookies, I guess.

Asking because Flask-Session is dead (https://github.com/fengsp/flask-session/issues/134, https://github.com/fengsp/flask-session/issues/128) and has lots of open issues/PRs.

**Comment by CyberTailor** [comment]
This issue is blocking adoption of the application in distro software repositories.

**Reply by CyberTailor** [reply]
@benbusby

**Reply by benbusby** [reply]
They're not *really* needed, but were the simplest way to achieve a more reliable state for users without relying on anything client side. It's been suggested in the past to switch to using localStorage for storing user config values, but I've been reluctant to implement that since it would break config options for anyone who blocks Javascript. I'm also not sure how well that would work with Opensearch. 

Not saying I'm not open to switching to a different approach/library/etc and moving away from sessions -- I'm very open to that actually. Flask-Session has given me more than enough headaches, so I'd love to move away from it at some point.
